src/libs/Network/SequentialQueue.js

```diff
@@ -68,6 +76,19 @@ function flush() {
                     isSequentialQueueRunning = false;
                     resolveIsReadyPromise();
                     currentRequest = null;
+
+                    // Resolve the isDonePromise once all the pending requests complete,
+                    // and reset the pendingRequests array
+                    Promise.all(pendingRequests).then(() => {
+                        resolveIsDonePromise();
+                        pendingRequests = [];
+                    });
+                }).then(() => {
+                    // Reset the isDonePromise so we can use it again on the next run,
+                    // this has to be done after the finally runs otherwise it'll have no effect
+                    isDonePromise = new Promise((resolve) => {
+                        resolveIsDonePromise = resolve;
+                    });
```

Comment:
Can we use the promise returned by `process()`  directly instead of add this other concept of `isDonePromise()`?

Reply:
It looks like the `.then()` of the promise returned here: 

```js
return task.then(process);
```

would be the same thing - but maybe I'm missing why that wouldn't work.

Reply:
@marcaaron I tried that and it appears to work only the first time. Any additional attempts and the following starts to happen. You can also see how doing this duplicates requests.

https://user-images.githubusercontent.com/31285285/199122488-d3269b7b-7411-4a1c-b8b2-9a3f4c7f45a3.mov

The promise returned from `process` doesn't actually wait for the requests on the queue to respond before resolving, it just waits for them all to finish firing. With that branch checked out, you can see that `UpdateWorkspaceCustomUnitRate`, a `write`, doesn't respond before `OpenWorkspaceReimburseView`, a `read`, is called. That needs to happen in order to guarantee the flickering effect does not happen.